pkg/runner/runner.go

```diff
@@ -170,7 +172,15 @@ func (runner *runnerImpl) NewPlanExecutor(plan *model.Plan) common.Executor {
 				}
 				pipeline = append(pipeline, common.NewParallelExecutor(maxParallel, stageExecutor...))
 			}
-			return common.NewParallelExecutor(runtime.NumCPU(), pipeline...)(ctx)
+			var ncpu int
+			info, err := container.GetHostInfo(ctx)
+			if err != nil {
+				log.Errorf("failed to obtain container engine info: %s", err)
+				ncpu = 1 // sane default?
```

Comment:
Or `runtime.NumCPU()`?
What's the benefit of asking the docker client for the number of CPUs? Does it take into account resource constraints via cgroups or the scheduler?

Reply:
> Or runtime.NumCPU()?

Once I contributed my hostexecutor, I would prefer:
- `runtime.NumCPU()` here, since I don't want to start the docker daemon to test something on the host.
- `failed to obtain container engine info: %s` should become a debug log, since docker would be optional

Until then, I'm fine with `1` as default and an error message.

> What's the benefit of asking the docker client for the number of CPUs?

A remote docker server might have a different amount of CPUs, but whatever. I never actually used a remote docker server.

Reply:
> What's the benefit of asking the docker client for the number of CPUs?

Docker host doesn't have to be on the same logical machine as `act`. Docker Desktop on macOS and Windows both run in VMs which can have different CPU count.